Mathlib/LinearAlgebra/TensorProduct/Graded.lean

```diff
@@ -0,0 +1,495 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.RingTheory.TensorProduct
+import Mathlib.RingTheory.GradedAlgebra.Basic
+import Mathlib.LinearAlgebra.DirectSum.TensorProduct
+import Mathlib.Data.ZMod.Basic
+import Mathlib.Algebra.CharP.Two
+
+/-!
+# Graded tensor products over super- (`ZMod 2`-graded)
+
+The graded tensor product $A \hat\otimes_R B$ is imbued with a multiplication defined on homogeneous
+tensors by:
+
+$$(a \otimes b) \cdot (a' \otimes b') = (-1)^{\deg a' \deg b} (a \cdot a') \otimes (b \cdot b')$$
+
+where $A$ and $B$ are algebras graded by `ZMod 2`, also known as superalgebras.
+
+## Main results
+
+* `TensorProduct.gradedMul`: this multiplication on externally-graded rings, as a bilinear map
+* `SuperTensorProduct R 𝒜 ℬ`: for families of submodules of `A` and `B` that form a graded algebra,
+  this is a type alias for `A ⊗'[R] B` with the appropriate multiplication.
+* `SuperTensorProduct.instAlgebra`: the ring structure induced by this multiplication.
+* `SuperTensorProduct.liftEquiv`: a universal property for graded tensor products
+
+
+## Notation
+
+`𝒜 ⊗'[R] ℬ` is notation for `SuperTensorProduct R 𝒜 ℬ`
+
+## References
+
+* https://math.stackexchange.com/q/202718/1896
+* TODO: find appropriate part of Bourbaki
+-/
+
+suppress_compilation
+
+local notation "ℤ₂" => ZMod 2
+open scoped TensorProduct
+
+variable {R A B : Type*}
+
+section zmod2_pow
+
+/-- There is a canonical power operation by `ℤˣ` on `ZMod 2`. -/
+instance : Pow ℤˣ (ZMod 2) where
+  pow s z₂ := s ^ z₂.val
+
+lemma z₂pow_def (s : ℤˣ) (x : ZMod 2) : s ^ x = s ^ x.val := rfl
+
+@[simp] lemma z₂pow_zero (s : ℤˣ) : (s ^ (0 : ZMod 2) : ℤˣ) = (1 : ℤˣ) := pow_zero _
+@[simp] lemma z₂pow_one (s : ℤˣ) : (s ^ (1 : ZMod 2) : ℤˣ) = s := pow_one _
+lemma z₂pow_add (s : ℤˣ) (x y : ZMod 2) : s ^ (x + y) = s ^ x * s ^ y := by
+  simp only [z₂pow_def]
+  rw [ZMod.val_add, ←pow_eq_pow_mod, pow_add]
+  fin_cases s <;> simp
+
+end zmod2_pow
```

Comment:
I've made [a zulip thread about this](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Powers.20of.20.60.E2.84.A4.CB.A3.60.20by.20.60ZMod.202.60/near/396291926).